app/src/main/java/com/github/se/eventradar/model/repository/event/MockEventRepository.kt

```diff
@@ -0,0 +1,63 @@
+package com.github.se.eventradar.model.repository.event
+
+import com.github.se.eventradar.model.Location
+import com.github.se.eventradar.model.Resource
+import com.github.se.eventradar.model.event.Event
+import com.github.se.eventradar.model.event.EventCategory
+import com.github.se.eventradar.model.event.EventTicket
+import java.time.LocalDateTime
+import javax.inject.Inject
+
+class MockEventRepository @Inject constructor() : IEventRepository {
+  override suspend fun getEvents(): Resource<List<Event>> {
+    val events =
+        List(20) {
+          Event(
+              eventName = "Event $it",
+              eventPhoto = "",
+              start = LocalDateTime.parse("2021-10-01T10:00:00"),
+              end = LocalDateTime.parse("2021-10-01T12:00:00"),
+              location = Location(0.0, 0.0, "Test Location"),
+              description = "Test Description",
+              ticket = EventTicket("Test Ticket", 0.0, 1),
+              contact = "Test Contact Email",
+              organiserList = setOf("Test Organiser"),
+              attendeeList = setOf("Test Attendee"),
+              category = EventCategory.COMMUNITY,
+              fireBaseID = "$it")
+        }
+
+    return Resource.Success(events)
+  }
```

Comment:
This is crucial! **This repository will** (ideally) be utilized in the tests (instead of making connections with firestore), even if you're conducting instrumented tests. During these instrumented tests, you'll probably need to add/remove events and verify if the UI responds accurately to these modifications. 

However, with your current implementation of `MockEventRepository`, this won't be feasible. I recommend transforming `MockEventRepository` into an in-memory database, where data is stored in a map. If you end up doing this, make sure you implement the other methods as well.

```suggestion
class MockEventRepository constructor() : IEventRepository {
  private val storage: MutableMap<String, Event> = mutableMapOf()

  override suspend fun getEvents(): Resource<List<Event>> {
    val events = storage.values.toList()
    return Resource.Success(events)
  }
```

> [!TIP]
> Take a look at the first 40 lines of [this file](https://github.com/RaulinN/swent-snippets/blob/migration-hilt-full-file/hilt-migration-from-manual-di.kt)

Note that you can obviously add default events in your map if you wish so